contrib/vagrant/provision-util.sh

```diff
@@ -144,29 +208,115 @@ os::util::install-sdn() {
       # origin multitenant plugin knows how to discover it.
       cp -f multitenant/bin/openshift-ovs-multitenant /usr/bin/
       cp -f multitenant/bin/openshift-sdn-multitenant-setup.sh /usr/bin/
+    popd > /dev/null
+  fi
 
-      # subnet and multitenant plugin setup writes docker network options
-      # to /run/openshift-sdn/docker-network, make this file to be exported
-      # as part of docker service start.
-      local system_docker_path="/usr/lib/systemd/system/docker.service.d/"
-      mkdir -p "${system_docker_path}"
-      cat <<EOF > "${system_docker_path}/docker-sdn-ovs.conf"
+  # subnet and multitenant plugin setup writes docker network options
+  # to /run/openshift-sdn/docker-network, make this file to be exported
+  # as part of docker service start.
+  local system_docker_path="/usr/lib/systemd/system/docker.service.d/"
+  mkdir -p "${system_docker_path}"
+  cat <<EOF > "${system_docker_path}/docker-sdn-ovs.conf"
 [Service]
 EnvironmentFile=-/run/openshift-sdn/docker-network
 EOF
-    popd > /dev/null
+
+  systemctl enable openvswitch &> /dev/null
+  systemctl start openvswitch
+}
+
+os::provision::base-provision() {
+  os::provision::fixup-net-udev
+
+  os::provision::setup-hosts-file "${MASTER_NAME}" "${MASTER_IP}" NODE_NAMES \
+    NODE_IPS
+
+  os::provision::install-pkgs
+}
+
+os::provision::fixup-net-udev() {
+  if [ "${FIXUP_NET_UDEV}" == "true" ]; then
+    NETWORK_CONF_PATH=/etc/sysconfig/network-scripts/
+    rm -f ${NETWORK_CONF_PATH}ifcfg-enp*
+    if [[ -f "${NETWORK_CONF_PATH}ifcfg-eth1" ]]; then
+      sed -i 's/^NM_CONTROLLED=no/#NM_CONTROLLED=no/' ${NETWORK_CONF_PATH}ifcfg-eth1
+      if ! grep -q "NAME=" ${NETWORK_CONF_PATH}ifcfg-eth1; then
+        echo "NAME=openshift" >> ${NETWORK_CONF_PATH}ifcfg-eth1
+      fi
+      nmcli con reload
+      nmcli dev disconnect eth1
+      nmcli con up "openshift"
+    fi
   fi
+}
+
+os::provision::in-container() {
+  test -f /.dockerinit
+}
+
+os::provision::install-pkgs() {
+  # Only install packages if not deploying to a container.  A
+  # container is expected to have installed packages as part of image
+  # creation.
+  if ! os::provision::in-container; then
+    yum update -y
```

Comment:
So my earlier comment got eaten up in wherever old github review comments go to die. Anyway, I'd said you should "yum -y install deltarpm" first, and you said "but isn't it already installed?"

    ==> dind-host: Total download size: 295 M
    ==> dind-host: Downloading packages:
    ==> dind-host: Delta RPMs disabled because /usr/bin/applydeltarpm not installed.

I know the non-dind deployment also hits this. Anyway, it doesn't hurt to try to just always try to install it.